### ISSUE TITLE
Color picker

### ISSUE THREAD

**Issue by probablykasper** [zed-industries/extensions#495]
### Check for existing issues

- [X] Completed

### Describe the feature

Having a good color picker is a important for me when working with e.g CSS.

Features:
- A command to open the color picker anywhere (not language-specific)
- Try to detect the color from the selected text
- Try to detect the color surrounding the cursor (if no text is selected)
- Allow changing the format (HEX/RGB/HSL)
- When finished, press `Enter` and the color is pasted in

**Comment by peterjskaltsis** [comment]
> I'd love it if, when parsing hex colors, it placed a tiny square with the color right before it.
> 
> However, a color picker popup from a command like this would work as well.

> https://github.com/nickpoorman/colord?tab=readme-ov-file#zed-integration

Agree, I used this in VSCode previously (works for more than just JSON files):
https://marketplace.visualstudio.com/items?itemName=yechunan.json-color-token

Would love if we could build extensions to contribute these sorts of things to the community

**Comment by jxcg** [comment]
Is there an extension yet for this?

**Comment by eatenpancreas** [comment]
Agreed,

In jetbrains ide's you're able to see the colors from a quick glance:
![image](https://github.com/user-attachments/assets/e48a6ee7-51e9-4f09-9e08-5cf373c93104)
Not sure if this feature is possible, but certainly one i'd love to see in zed

**Comment by Vishtar** [comment]
For a reference: https://marketplace.visualstudio.com/items?itemName=naumovs.color-highlight
This extension can color background and etc. of a hex/rgb.

**Comment by elyalon** [comment]
Suggestion: Support an OKLCH interface.

I've been using [this standalone OKLCH color picker](https://github.com/eero-lehtinen/oklch-color-picker) and it feels much more accurate than the standard HSL pickers (like the one in VSCode) when doing modifications on a color. It's also written in Rust so it could be useful as a reference.
